### DISCUSSION TITLE
Tutorial - buttons not centered

### DISCUSSION THREAD

**Discussion by dswisher** [AvaloniaUI/Avalonia#7884] [Q&A] [answered]
I'm new to Avalonia, working through the [ToDo Tutorial](https://docs.avaloniaui.net/tutorials/todo-list-app/adding-new-items). The screen shot there (and in earlier sections) shows buttons stretched across the width of the `DockPanel`, with centered text:

![Screen Shot 2022-03-27 at 1 03 42 PM](https://user-images.githubusercontent.com/78451/160294533-0b774331-fd1a-42c0-8f05-e0a7872fdb3e.png)

When I use the same XAML, the buttons appear on the left (not stretched).

![Screen Shot 2022-03-27 at 1 04 38 PM](https://user-images.githubusercontent.com/78451/160294601-9ab44dc1-ffd5-4862-9627-e7d9ccbd4eb3.png)

If I apply `HorizontalAlignment="Stretch"` to one of the buttons, it does indeed stretch, but the text is not centered.

![Screen Shot 2022-03-27 at 1 06 26 PM](https://user-images.githubusercontent.com/78451/160294649-8c279278-f3a1-4657-8f57-4e7fbafa4c2c.png)

How can I match what the tutorial shows?

The project was created with the latest `dotnet new` template (a few days ago). I'm not yet done with the tutorial, but I've pushed the code to a github repo: [avalonia-todo-tut](https://github.com/dswisher/avalonia-todo-tut/tree/224158ea541fd1b91fbb9574d1dc0af6f95172b0). The [AddItemView.axaml](https://github.com/dswisher/avalonia-todo-tut/blob/224158ea541fd1b91fbb9574d1dc0af6f95172b0/Views/AddItemView.axaml) _seems_ straightforward:

```
<UserControl xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
             xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
             mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
             x:Class="MyApp.Views.AddItemView">
    <DockPanel>
        <Button DockPanel.Dock="Bottom">Cancel</Button>
        <Button DockPanel.Dock="Bottom" HorizontalAlignment="Stretch">OK</Button>
        <TextBox AcceptsReturn="true"
                 Text="{Binding Description}"
                 Watermark="Enter your TODO"></TextBox>
    </DockPanel>
</UserControl>
```

Thanks in advance for any help you can provide!

**Comment by robloo** [comment, marked as answer]
In WPF, `HorizontalAlignment` and `VerticalAlignment` are automatically set to `Stretch`. This ensures the buttons take all available space. I believe the Default/Simple theme in Avalonia also set these properties to stretch which is where the tutorial originated. However, now Fluent theme is standard and these default to `Left` and `Top` respectively so you will need to change them both to `Stretch` as you discovered.

There are two more properties that control the text (content) position `HorizontalContentAlignment` and `VerticalContentAlignment`. Set these to `Center` to get the same appearance as the tutorial. Personally, I would prefer if content was centered by default #7839

**Reply by dswisher** [reply]
That did the trick. Thank you for the quick response!
